_rules/element-lang-programmatically-determinable-7ed469.md

```diff
@@ -0,0 +1,142 @@
+---
+id: 7ed469
+name: Element language is programmatically determinable
+rule_type: atomic
+description: |
+  This rule checks that the language of an element in the page body can be programmatically determined.
+accessibility_requirements: # Remove whatever is not applicable
+  wcag20:3.1.2: # Language of Parts (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing is needed
+    inapplicable: further testing is needed
+input_aspects:
+  -  DOM Tree
+  -  CSS Styling
+
+acknowledgements:
+  authors:
+    -  Patrick H. Lauke
+---
+
+## Applicability
+
+This rules applies to any HTML element that:
+
+* contains a [text node](https://dom.spec.whatwg.org/#text)
+* is a descendant in the flat tree of a `body` element
+* is included in the accessibility tree
+
+## Expectation (1)
+
+For each test target, the test target itself or any of its ancestor elements (up to and including the root element of the document) has a `lang` attribute with a valid language subtag.
+
+## Assumptions
+
+_There are currently no assumptions_
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+In practice, this rule will be satisfied if the separate success criterion 3.1.1 Language of Page (Level: A) is satisfied. This rule ensures that if 3.1.1 is *not* satisfied, content is checked for alternative ways in which language of page content itself can still have a programmatically determinable human language.
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Language of the `p` element is inherited by the `lang` attribute on the `html` root element.
+
+```html
+<html lang="en">
+	<body>
+		<p>Content</p>
+	</body>
+</html>
+```
+
+#### Passed Example 2
+
+Language of the `p` element is inherited by the `lang` attribute on the `div` parent element. Note that this example fails 3.1.1 Language of Page (Level: A) as the `html` root element lacks a `lang` or `xml:lang` attribute.
+
+```html
+<html>
+	<body>
+		<div lang="en">
+			<p>Content</p>
+		</div>
+	</body>
+</html>
+```
+
+#### Passed Example 3
+
+Language of the `p` element is inherited by the `lang` attribute on the `div` ancestor element. Note that this example fails 3.1.1 Language of Page (Level: A) as the `html` root element lacks a `lang` or `xml:lang` attribute.
+
+```html
+<html>
+	<body>
+		<div lang="en">
+			<div>
+				<p>Content</p>
+			</div>
+		</div>
+	</body>
+</html>
+```
+
+### Failed
+
+#### Failed Example 1
+
+Language of the `p` element cannot be programmatically determined as the element itself and none of its ancestors, including the `html` root element, has a defined `lang` or `xml:lang` attribute. Note that this example also fails 3.1.1 Language of Page (Level: A) as the `html` root element lacks a `lang` or `xml:lang` attribute.
```

Comment:
I think we should remove all references to `xml:lang` to follow what we did we the other language rules.

Reply:
makes sense, this was already suggested by a colleague - i only included it here because there's still one existing rule that does mention `xml:lang` https://act-rules.github.io/rules/5b7ae0

Reply:
Yes. It's the rule that specifically test `xml:lang` 😄 
Other language rules were also using `xml:lang` but that has been cleaned recently (not that you can guess, of course…) Notably because it seems that `xml:lang` is almost not used in practice.
Additionally, your rule specifically requires a `lang` attribute (not an `xml:lang` one), so there no reason to mention `xml:lang`.